chore(curriculum): add two labs to full stack/javascript/loops (#60185)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lab-chunky-monkey/a9bd25c716030ec90084d8a1.md

---
id: a9bd25c716030ec90084d8a1
title: Implement the Chunky Monkey Algorithm
challengeType: 26
dashedName: implement-the-chunky-monkey-algorithm
---

# --description--

Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

> 1. Write a function named `chunkArrayInGroups` that splits an array (first argument) into groups the length of `size` (second argument) and returns them as a two-dimensional array.

Reviewer: ```suggestion
1. Create a function named `chunkArrayInGroups` that takes an array (`arr`) and a number (`size`) argument. It should split the array into chunks of the given size and return them as a two-dimensional array.
```

Reviewer 2: ```suggestion
1. Write a function named `chunkArrayInGroups` that takes an array as first argument and a number as second argument. The function should split the array into smaller arrays of length equal to the second argument and returns them as a two-dimensional array.
```

as we have removed the seed code, the campers are free to use what they want as parameters names.

---
id: a9bd25c716030ec90084d8a1
title: Chunky Monkey
challengeType: 26

> dashedName: chunky-monkey-algorithm

Reviewer 2: ```suggestion
dashedName: implement-the-chunky-monkey-algorithm
```

---

# --description--

> Write a function that splits an array (first argument) into groups the length of `size` (second argument) and returns them as a two-dimensional array.

Reviewer 3: ```suggestion
Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. Write a function named `chunkArrayInGroups` that splits an array (first argument) into groups the length of `size` (second argument) and returns them as a two-dimensional array.
```

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5], 3), [
  [0, 1, 2],
  [3, 4, 5]
]);
```

`chunkArrayInGroups([0, 1, 2, 3, 4, 5], 2)` should return `[[0, 1], [2, 3], [4, 5]]`.

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5], 2), [
  [0, 1],
  [2, 3],
  [4, 5]
]);
```

`chunkArrayInGroups([0, 1, 2, 3, 4, 5], 4)` should return `[[0, 1, 2, 3], [4, 5]]`.

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5], 4), [
  [0, 1, 2, 3],
  [4, 5]
]);
```

`chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6], 3)` should return `[[0, 1, 2], [3, 4, 5], [6]]`.

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6], 3), [
  [0, 1, 2],
  [3, 4, 5],
  [6]
]);
```

`chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6, 7, 8], 4)` should return `[[0, 1, 2, 3], [4, 5, 6, 7], [8]]`.

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6, 7, 8], 4), [
  [0, 1, 2, 3],
  [4, 5, 6, 7],
  [8]
]);
```

`chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6, 7, 8], 2)` should return `[[0, 1], [2, 3], [4, 5], [6, 7], [8]]`.

```js
assert.deepEqual(chunkArrayInGroups([0, 1, 2, 3, 4, 5, 6, 7, 8], 2), [
  [0, 1],
  [2, 3],
  [4, 5],
  [6, 7],
  [8]
]);
```

# --seed--

## --seed-contents--

```js

> function chunkArrayInGroups(arr, size) {
>   return arr;
> }
> 
> chunkArrayInGroups(['a', 'b', 'c', 'd'], 2);

Reviewer 3: ```suggestion
```
I think we can start with no seed code
